src/commcare_cloud/ansible/roles/pgbouncer/tasks/main.yml

```diff
@@ -44,11 +35,31 @@
     reload: yes
   with_dict: "{{ pgbouncer_kernel_settings }}"
 
-- name: pgbouncer init script
-  template: src=pgbouncer.init.j2 dest=/etc/init.d/pgbouncer
+- name: Clean up possible previous instances
+  shell: |
+    systemctl stop pgbouncer-multiprocess@*
+    systemctl disable pgbouncer-multiprocess@*
+    rm -rf {{ pgbouncer_config_home }}/pgbouncer-*.ini
+    rm -rf {{ pgbouncer_config_home }}/userlist-*.txt
+    rm -rf /var/run/postgresql/pgbouncer-*
```

Comment:
This looks like it will mean you cannot run this playbook while pgbouncer is running (since it will blow away the current config and socket directory). What happens if we want to increase the number of processes on a machine from 1 to 2 rather than start from scratch?

Reply:
Right. I mean, you can, but of course it would be a hard transition. I'm only leaving the logs, because they could be of value.
This part can definitely be improved/expanded, let's define some requirements and I'll work on a better solution.

Reply:
Is the goal here to ensure that we don't have dangling config if we reduce the number of processes? If so we could delete all config from `pgbouncer_processes` up to some max number like 32 or 64? Or else we could use the same approach we use for supervisor configs: https://github.com/dimagi/commcare-cloud/blob/16f055deef97ca4795bd2fb9ee3cc08d0e7fedcb/src/commcare_cloud/ansible/deploy_commcarehq.yml#L97-L128

Reply:
Yes, but also to be able to scale up or down without restarting everything from scratch every time. The new implementation should be much more clean and effective, let me know if you think we could still move some parts of it elsewhere.